doc/man3/BIO_s_mem.pod

```diff
@@ -104,6 +105,12 @@ BIO is set to BIO_NOCLOSE, before freeing the BUF_MEM the data pointer
 in it must be set to NULL as the data pointer does not point to an
 allocated memory.
 
+Calling BIO_reset() on a read write memory BIO with BIO_FLAGS_NONCLEAR_RST
+flag set can have unexpected outcome when the reads and writes to the
+BIO are intertwined. As documented above the BIO will be reset to the
+state after the last completed write operation. The effects of reads
+preceeding that write operation cannot be undone.
```

Comment:
I assume that BIO_get_mem_ptr changes the reset state, right?

Reply:
Good point, I've added a paragraph documenting it.